woocommerce/wc-admin
# Move capability checks to client

Fixes #6315

Moves all the privilege checks to client-side and loads WooCommerce Admin for all users.

This adds permission checks to:
* Activity Panel tabs
* WCA Routes
* Menu items

Follow-ups are probably needed to add capability checks for:
* The setup wizard
* Task list
* Inbox notes

### Screenshots

<img width="1046" alt="Screen Shot 2021-02-17 at 6 41 44 PM" src="https://user-images.githubusercontent.com/10561050/108282877-5b6b1980-7150-11eb-910b-abdc6333cbb9.png">


### Detailed test instructions:

1. Create various non-admin users with custom capabilities.  Make sure to not include the `view_woocommerce_reports` for at least one role.  https://wordpress.org/plugins/leira-roles/
2. Log in as the non-admin users.
3. Check that the correct menu items are shown.
4. Check that there aren't items shown to the user they should not be able to use or interact with.
5. Enable the new navigation under WooCommerce -> Settings -> Advanced -> Features.
6. Check that the users are able to see the new navigation menu.
